### PR TITLE
Fix ExcessMapEstimator to account for mask safe

### PR DESCRIPTION
**Pull request by QRemy**
Fix ExcessMapEstimator to account for mask safe

### CODE REVIEW

File: gammapy/estimators/map/tests/test_excess.py [hunk 1]
```diff
@@ -142,6 +154,27 @@ def test_significance_map_estimator_map_dataset(simple_dataset):
     assert_allclose(result["npred_excess_ul"].data[0, 10, 10], 107.806275, atol=1e-3)
 
 
+def test_significance_map_estimator_map_dataset_mask_safe(simple_dataset_mask_safe):
+    simple_dataset_mask_safe.exposure = None
+    energy_edges = simple_dataset_mask_safe.counts.geom.axes[0].edges
+    energy_edges = [energy_edges[0], energy_edges[-1]]
+    estimator = ExcessMapEstimator(
+        0.1 * u.deg, selection_optional=["all"], energy_edges=energy_edges
+    )
```
Comment (registerrier) [inline gammapy/estimators/map/tests/test_excess.py:163]:
```suggestion
    estimator = ExcessMapEstimator(0.1 * u.deg, selection_optional=["all"])
```
If you take min and max edges, you don't need to specify them.

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy . Looks good.

Should we use the `mask_safe` or the full `mask`?

**Comment by QRemy** [comment]
> Should we use the `mask_safe` or the full `mask`?

It should be consistent with `MapDataset.resample_energy_axis` that apply only mask_safe.
If we change the behaviour of `MapDataset.resample_energy_axis` to include the mask_fit by default it should probably done in a separate PR. I don't have a strong opinion about that personally I just redefine the mask_safe when needed.

**Review by AtreyeeS — APPROVED** [review]
Thanks @QRemy ! 

I think in a future PR, it will be good to take the fit mask into account.
Usually, we discourage users from modifying the mask safe.

**Comment by registerrier** [comment]
Indeed in practice, we usually need an exclusion mask to compute the excess maps. Using the `mask_fit` is likely the simpler approach.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy . No further comment.

**Pull request merged**